lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb

```diff
@@ -1,56 +1,122 @@
+          return "#{vp[MAJOR_NUMBER]}.#{vp[MINOR_NUMBER]}" unless is_hotfix?(version)
           "#{vp[MAJOR_NUMBER]}.#{vp[MINOR_NUMBER]}.#{vp[HOTFIX_NUMBER]}"
         end
 
+        # Extract the version name and code from the `vanilla` flavor of the `$PROJECT_NAME/build.gradle file`
+        #   or for the defaultConfig if `HAS_ALPHA_VERSION` is not defined.
+        #
+        # @env HAS_ALPHA_VERSION If set (with any value), indicates that the project uses `vanilla` flavor.
+        #
+        # @return [Hash] A hash with 2 keys "name" and "code" containing the extracted version name and code, respectively
+        #
         def self.get_release_version
           section = ENV["HAS_ALPHA_VERSION"].nil? ? "defaultConfig" : "vanilla {"
           gradle_path = self.gradle_path
-          name = get_version_name_from_file(gradle_path, section)
-          code = get_version_build_from_file(gradle_path, section)
+          name = get_version_name_from_gradle_file(gradle_path, section)
+          code = get_version_build_from_gradle_file(gradle_path, section)
           return { VERSION_NAME => name, VERSION_CODE => code }
         end
 
+        # Extract the version name and code from the `defaultConfig` of the `$PROJECT_NAME/build.gradle` file
+        #
+        # @return [Hash] A hash with 2 keys `"name"`` and `"code"`` containing the extracted version name and code, respectively,
+        #                or `nil` if `$HAS_ALPHA_VERSION` is not defined.
+        #
         def self.get_alpha_version
-          if (ENV["HAS_ALPHA_VERSION"].nil?)
-            return nil
-          end
+          return nil if ENV["HAS_ALPHA_VERSION"].nil?
 
           section = "defaultConfig"
           gradle_path = self.gradle_path
-          name = get_version_name_from_file(gradle_path, section)
-          code = get_version_build_from_file(gradle_path, section)
+          name = get_version_name_from_gradle_file(gradle_path, section)
+          code = get_version_build_from_gradle_file(gradle_path, section)
           return { VERSION_NAME => name, VERSION_CODE => code }
         end
   
-        def self.is_alpha_version(version)
+        # Determines if a version name corresponds to an alpha version (starts with `"alpha-"`` prefix)
+        #
+        # @param [String] version The version name to check
+        #
+        # @return [Bool] true if the version name starts with the `ALPHA_PREFIX`, false otherwise.
+        #
+        #private
+        def self.is_alpha_version?(version)
           version[VERSION_NAME].start_with?(ALPHA_PREFIX)
         end
   
-        def self.is_beta_version(version)
+
+        # Check if this versionName corresponds to a beta, i.e. contains some `-rc` suffix
+        #
+        # @param [String] version The versionName string to check for
+        #
+        # @return [Bool] True if the version string contains `-rc`, indicating it is a beta version.
+        #
+        #public
```

Comment:
Same thing here, though not sure if this is correct or not?

```suggestion
        # @public
```

Reply:
Tbh this are more remnants of personal/debug annotations to mark for myself which methods I explicitly and manually checked to see if they were called from outside of this file or not, and could later be put inside a `private` section when I will reorder the methods in the file (because in Ruby the `private` keyword applies to _all_ the methods that follow it until the end of the file/module/scope so we have to groupe them all at the end of the scope; something I plan to do in a future, separate PR)